alot/commands/envelope.py

```diff
@@ -392,93 +392,72 @@ def openEnvelopeFromTmpfile():
         await ui.apply_command(cmd)
 
 
-@registerCommand(MODE, 'set', arguments=[
-    (['--append'], {'action': 'store_true', 'help': 'keep previous values'}),
+@registerCommand(MODE, 'set', forced={'action': 'set'}, arguments=[
+    (['--append'],
+     {'action': 'store_true', 'help': 'keep previous values'}),
+    (['key'], {'help': 'header to refine'}),
+    (['value'], {'nargs': '+', 'help': 'value'})])
+@registerCommand(MODE, 'unset', forced={'action': 'unset'}, arguments=[
+    (['key'], {'help': 'header to refine'})])
+@registerCommand(MODE, 'toggleset', forced={'action': 'toggleset'}, arguments=[
     (['key'], {'help': 'header to refine'}),
     (['value'], {'nargs': '+', 'help': 'value'})])
 class SetCommand(Command):
-    """set header value"""
-    def __init__(self, key, value, append=False, **kwargs):
+    """set/unset/toggle a header value"""
+    repeatable = True
+
+    def __init__(self, action, key, value=None, append=False, **kwargs):
         """
+        :param action: whether to set/unset/toggleset
+        :type action: str
         :param key: key of the header to change
         :type key: str
         :param value: new value
         :type value: str
         """
+        self.action = action
         self.key = key
-        self.value = ' '.join(value)
+
+        # a value is only needed if we set or toggle else it is None
+        self.value = ' '.join(value) if value is not None else None
+
         self.reset = not append
         Command.__init__(self, **kwargs)
 
     async def apply(self, ui):
         envelope = ui.current_buffer.envelope
-        if self.reset:
+
```

Comment:
I'd prefer to do it without the extra `do_set` variable and sinmply make changes to the envelope within the respective clauses to keep things simple.

Reply:
This is fixed in https://github.com/pazz/alot/pull/1315/commits/ca0bef0067646f94cc2f7b18391cc6f23e128f73